### DISCUSSION TITLE
No module named speech_recognition

### DISCUSSION THREAD

**Discussion by khaledgomaa** [exodrifter/unity-python#4] [Q&A]
I got speech_recognition library from my python installation and copied it to Lib path but it doesn't work too and i tried to put the path of my python library installation but it doesn't work too , one more thing what is
string code = @"
import os
filename = os.path.abspath ('test.txt');";  ?

to install speech_recognition : pip install SpeechRecognition
to install pyaudio : pip install pyaudio

you can try it.

**Comment by exodrifter** [comment]
```
string code = @"
import os
filename = os.path.abspath ('test.txt');";
```
This snippet of code is just example code to show the python library is working.

After installing `speech_recognition` and `pyaudio` using pip, I copied the respective files to the Lib path and it worked, so you are probably doing something wrong. Please make sure you using these lines, which set the search paths for IronPython to find python libraries to use.
```
var paths = engine.GetSearchPaths();
paths.Add(Application.dataPath + "/Python/Lib");
engine.SetSearchPaths (paths);
```
Make sure the paths you are providing are valid and correct. If it still does not work, please post an example that reproduces the error.

**Comment by khaledgomaa** [comment]
I tried a lot but it doesn't work too , could you upload the project after adding my code and lib needed if it possible , I'm sorry for that really
thanks in advance.

**Comment by exodrifter** [comment]
I'm sorry, I have other work to do and I don't have time to do your work for you.

I will reopen this ticket if you post your project or your code that reproduces the problem.
